script/GameObjects/Elements/Core/Damage.lua

```diff
@@ -1,31 +1,54 @@
 local Entity = require('GameObjects.Entity')
-local Bindings = require('States.ApplicationBindings')
-
-function Entity:addHealth (max, rate)
-  assert(not self.health)
-  assert(max)
-  assert(rate)
-  self.health = max
-  self.healthMax = max
-  self.healthRate = rate
-  self:register(Event.Update, Entity.updateHealth)
-end
 
-function Entity:damage (amount, source)
-  assert(self.health)
+function Entity:applyDamage (amount, source)
+  local damageRemaining = amount
+  local shieldRemaining = self:mgrShieldGetShield()
+  local armorRemaining  = self:mgrArmorGetArmor()
 
-  if self.health <= 0 then return end
+local thisShipName      = self:getName()
```

Comment:
intendation

Reply:
I'll check this.

Reply:
This was left unindented to signal that it's a temporary solution for putting "Ace" in the displayed name of more capable NPC ships. (Currently "ace" means an NPC ship can use Boost, which gives it a significant increase in maneuverability and makes it a tougher opponent in dogfighting.)

I have indented that piece of code and added comments to explain it.